pydantic/fields.py

```diff
@@ -69,6 +69,7 @@ def __init__(
 
         self.name: str = name
         self.alias: str = alias or name
+        self.pun_alias: bool = alias == name
```

Comment:
I felt adding a new field as the least intrusive solution.

The only real use-case for this field is to support settings which need to support pun-aliases to circumvent its default behaviour of:

- using var prefix
- using uppercase

Therefore maybe a comment is in order too, that explains such.